commons/zenoh-protocol-core/src/key_expr/borrowed.rs

```diff
@@ -249,11 +265,12 @@ impl keyexpr {
     pub unsafe fn from_slice_unchecked(s: &[u8]) -> &Self {
         std::mem::transmute(s)
     }
-    pub fn new<'a, T, E>(t: T) -> Result<&'a Self, E>
-    where
-        &'a Self: TryFrom<T, Error = E>,
-    {
-        t.try_into()
+}
+
+impl std::ops::Div for &keyexpr {
+    type Output = OwnedKeyExpr;
+    fn div(self, rhs: Self) -> Self::Output {
+        self.join(rhs).unwrap() // Joining 2 key expressions should always result in a canonizable string.
```

Comment:
Didn't you forget the call to `Self::autocanonize(s).unwrap()` here (as done in Div for OwnedKeyExpr) ?

Reply:
I didn't, `join` already does autocanonization, it just doesn't assume that things will work out as we can with a pair or KEs